test/image/mocks/domain_ref_axis_types.json

```diff
@@ -0,0 +1,1767 @@
+        "y0": 0,
+        "y1": 1,
+        "yref": "y7 domain"
+      },
+      {
+        "type": "line",
+        "x0": 0,
+        "x1": 1,
+        "xref": "x8 domain",
+        "y0": 172800000,
+        "y1": 172800000,
+        "yref": "y8"
+      },
+      {
+        "type": "line",
+        "x0": 0,
+        "x1": 1,
+        "xref": "x8 domain",
+        "y0": 259200000,
+        "y1": 259200000,
+        "yref": "y8"
+      },
+      {
+        "type": "line",
+        "x0": 2,
+        "x1": 2,
+        "xref": "x8",
+        "y0": 0,
+        "y1": 1,
+        "yref": "y8 domain"
+      },
+      {
+        "type": "rect",
+        "x0": 0,
+        "x1": 1,
+        "xref": "x8 domain",
+        "y0": 0,
+        "y1": 86400000,
+        "yref": "y8"
+      },
+      {
+        "type": "rect",
+        "x0": 3,
+        "x1": 5,
+        "xref": "x8",
+        "y0": 0,
+        "y1": 1,
+        "yref": "y8 domain"
+      },
+      {
+        "type": "line",
+        "x0": 0,
+        "x1": 1,
+        "xref": "x9 domain",
+        "y0": "bear",
+        "y1": "bear",
+        "yref": "y9"
+      },
+      {
+        "type": "line",
+        "x0": 0,
+        "x1": 1,
+        "xref": "x9 domain",
+        "y0": "alex",
+        "y1": "alex",
+        "yref": "y9"
+      },
+      {
+        "type": "line",
+        "x0": 2,
+        "x1": 2,
+        "xref": "x9",
+        "y0": 0,
+        "y1": 1,
+        "yref": "y9 domain"
+      },
+      {
+        "type": "rect",
+        "x0": 0,
+        "x1": 1,
+        "xref": "x9 domain",
+        "y0": "dog",
+        "y1": "cat",
+        "yref": "y9"
+      },
+      {
+        "type": "rect",
+        "x0": 3,
+        "x1": 5,
+        "xref": "x9",
+        "y0": 0,
+        "y1": 1,
+        "yref": "y9 domain"
+      },
+      {
+        "type": "line",
+        "x0": 0,
+        "x1": 1,
+        "xref": "x10 domain",
+        "y0": [
+          "B",
+          "piano"
+        ],
+        "y1": [
+          "B",
+          "piano"
+        ],
+        "yref": "y10"
+      },
+      {
+        "type": "line",
+        "x0": 0,
+        "x1": 1,
+        "xref": "x10 domain",
+        "y0": [
+          "C",
+          "mojtaba"
+        ],
+        "y1": [
+          "C",
+          "mojtaba"
+        ],
+        "yref": "y10"
+      },
+      {
+        "type": "line",
+        "x0": 2,
+        "x1": 2,
+        "xref": "x10",
+        "y0": 0,
+        "y1": 1,
+        "yref": "y10 domain"
+      },
+      {
+        "type": "rect",
+        "x0": 0,
+        "x1": 1,
+        "xref": "x10 domain",
+        "y0": [
+          "A",
+          "table"
+        ],
+        "y1": [
+          "B",
+          "chair"
+        ],
+        "yref": "y10"
+      },
+      {
+        "type": "rect",
+        "x0": 3,
+        "x1": 5,
+        "xref": "x10",
+        "y0": 0,
+        "y1": 1,
+        "yref": "y10 domain"
+      }
+    ],
+    "template": {
+      "data": {
```

Comment:
Please remove any unused template setup from this mock.

Reply:
Could we add a git hook like in Plotly.py?